lib/mongo/uri.rb

```diff
@@ -46,6 +46,39 @@ class URI
     # @since 2.0.0
     attr_reader :servers
 
+    # The mongodb connection string scheme.
+    #
+    # @deprecated Will be removed in 3.0.
+    #
+    # @since 2.0.0
+    SCHEME = 'mongodb://'.freeze
+
+    # The mongodb connection string scheme root.
+    #
+    # @since 2.5.0
+    MONGODB_SCHEME = 'mongodb'.freeze
+
+    # The mongodb srv protocol connection string scheme root.
+    #
+    # @since 2.5.0
+    MONGODB_SRV_SCHEME = 'mongodb+srv'.freeze
+
+    # Error details for an invalid scheme.
+    #
+    # @since 2.1.0
+    INVALID_SCHEME = "Invalid scheme. Scheme must be '#{MONGODB_SCHEME}' or '#{MONGODB_SRV_SCHEME}'".freeze
+
+    # MongoDB URI format specification.
+    #
+    # @since 2.0.0
+    FORMAT = 'mongodb://[username:password@]host1[:port1][,host2[:port2]' +
```

Comment:
Does the scheme need to reflect the possible `+srv` here?

Reply:
yes it should, thank you.

Reply:
I've updated the SRVProtocol class to create its own InvalidURI errors so it can pass in its own format info.